### ISSUE TITLE
Fastlane build failing Exit status of command 'bundle exec pod install' was 1 instead of 0

### ISSUE THREAD

**Issue by jlubeck** [fastlane/fastlane#9839]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

```
[17:22:37]: -------------------------------------------------
[17:22:37]: --- Step: Verifying required fastlane version ---
[17:22:37]: -------------------------------------------------
[17:22:37]: Your fastlane version 2.48.0 matches the minimum requirement of 1.101.0  ✅
[17:22:37]: ------------------------------
[17:22:37]: --- Step: default_platform ---
[17:22:37]: ------------------------------
[17:22:37]: Driving the lane 'ios' 🚀
[17:22:37]: -------------------------------
[17:22:37]: --- Step: ensure_git_branch ---
[17:22:37]: -------------------------------
[17:22:37]: Git branch match `master`, all good! 💪
[17:22:41]: -----------------------
[17:22:41]: --- Step: cocoapods ---
[17:22:41]: -----------------------
[17:22:42]: $ bundle exec pod install --repo-update
[17:22:42]: ▸ Installing dependencies of a pod spec:
[17:22:42]: ▸ $ pod install [NAME] [PROJECT]
[17:22:42]: ▸ Downloads all dependencies of the specified podspec file `NAME',
[17:22:42]: ▸ creates an Xcode Pods library project in `./Pods', and sets up `PROJECT'
[17:22:42]: ▸ to use the specified pods (if `PROJECT' is given). In case `NAME' is
[17:22:42]: ▸ omitted it defaults to either `Podfile' or `*.podspec' in the current
[17:22:42]: ▸ working directory.
[17:22:42]: ▸ Options
[17:22:42]: ▸ -------
[17:22:42]: ▸ --no-clean  Leave SCM dirs like `.git' and `.svn' in tact after downloading
[17:22:42]: ▸ --help      Show help information
[17:22:42]: ▸ --silent    Print nothing
[17:22:42]: ▸ --verbose   Print more information while working
[17:22:42]: ▸ --version   Prints the version of CocoaPods
[17:22:43]: ---------------------
[17:22:43]: --- Step: mailgun ---
[17:22:43]: ---------------------
+------------------------------------+---------------------------------------+
|                                Lane Context                                |
+------------------------------------+---------------------------------------+
| DEFAULT_PLATFORM                   | ios                                   |
| PLATFORM_NAME                      | ios                                   |
| LANE_NAME                          | ios                               |
| SIGH_PROFILE_TYPE                  | enterprise                            |
| MATCH_PROVISIONING_PROFILE_MAPPING | {"com..ios"=>"match InHouse  |
|                                    | com..ios"}                   |
+------------------------------------+---------------------------------------+
[17:22:44]: Exit status of command 'bundle exec pod install --repo-update' was 1 instead of 0.
Installing dependencies of a pod spec:

    $ pod install [NAME] [PROJECT]

      Downloads all dependencies of the specified podspec file `NAME',
      creates an Xcode Pods library project in `./Pods', and sets up `PROJECT' 
      to use the specified pods (if `PROJECT' is given). In case `NAME' is
      omitted it defaults to either `Podfile' or `*.podspec' in the current
      working directory.


Options
-------

    --no-clean  Leave SCM dirs like `.git' and `.svn' in tact after downloading
    --help      Show help information
    --silent    Print nothing
    --verbose   Print more information while working
    --version   Prints the version of CocoaPods


+------+------------------------------+-------------+
|                 fastlane summary                  |
+------+------------------------------+-------------+
| Step | Action                       | Time (in s) |
+------+------------------------------+-------------+
| 1    | Verifying required fastlane  | 0           |
|      | version                      |             |
| 2    | default_platform             | 0           |
| 3    | ensure_git_branch            | 0           |
| 4    | match                        | 4           |
| 💥    | cocoapods                    | 0           |
| 6    | mailgun                      | 1           |
+------+------------------------------+-------------+

[17:22:44]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

/Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane_core/lib/fastlane_core/ui/interface.rb:143:in `shell_error!': [!] Exit status of command 'bundle exec pod install --repo-update' was 1 instead of 0. (FastlaneCore::Interface::FastlaneShellError)
Installing dependencies of a pod spec:

    $ pod install [NAME] [PROJECT]

      Downloads all dependencies of the specified podspec file `NAME',
      creates an Xcode Pods library project in `./Pods', and sets up `PROJECT' 
      to use the specified pods (if `PROJECT' is given). In case `NAME' is
      omitted it defaults to either `Podfile' or `*.podspec' in the current
      working directory.


Options
-------

    --no-clean  Leave SCM dirs like `.git' and `.svn' in tact after downloading
    --help      Show help information
    --silent    Print nothing
    --verbose   Print more information while working
    --version   Prints the version of CocoaPods

	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane_core/lib/fastlane_core/ui/ui.rb:14:in `method_missing'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/helper/sh_helper.rb:56:in `sh_control_output'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/helper/sh_helper.rb:10:in `sh'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/actions/cocoapods.rb:27:in `run'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/runner.rb:252:in `block (2 levels) in execute_action'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/runner.rb:230:in `block in execute_action'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/runner.rb:226:in `chdir'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/runner.rb:226:in `execute_action'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/runner.rb:148:in `trigger_action_by_name'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/fast_file.rb:146:in `method_missing'
	from Fastfile:206:in `block (2 levels) in parsing_binding'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/lane_manager.rb:52:in `cruise_lane'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/command_line_handler.rb:30:in `handle'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/commands_generator.rb:104:in `block (2 levels) in run'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:178:in `call'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:153:in `run'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/runner.rb:476:in `run_active_command'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:64:in `run!'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/delegates.rb:15:in `run!'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/commands_generator.rb:303:in `run'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/fastlane/lib/fastlane/cli_tools_distributor.rb:66:in `take_off'
	from /Users/jan/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.48.0/bin/fastlane:20:in `<top (required)>'
	from /Users/jan/.rbenv/versions/2.3.1/bin/fastlane:22:in `load'
	from /Users/jan/.rbenv/versions/2.3.1/bin/fastlane:22:in `<main>'
```
### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.5                                     |
| Ruby                        | 2.3.1                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.11.0 (Apple Git-81)           |
| Installation Source         | ~/.rbenv/versions/2.3.1/bin/fastlane        |
| Host                        | Mac OS X 10.12.5 (16F73)                    |
| Ruby Lib Dir                | ~/.rbenv/versions/2.3.1/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2j  26 Sep 2016                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.3                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.48.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| io-console                | 0.4.5        |
| CFPropertyList            | 2.3.5        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| aws-sigv4                 | 1.0.1        |
| jmespath                  | 1.3.1        |
| aws-sdk-core              | 2.10.14      |
| aws-sdk-resources         | 2.10.14      |
| aws-sdk                   | 2.10.14      |
| babosa                    | 1.0.2        |
| bundler                   | 1.14.2       |
| claide                    | 1.0.2        |
| cocoapods                 | 0.2.0        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| highline                  | 1.7.8        |
| commander                 | 4.4.3        |
| commander-fastlane        | 4.4.5        |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| dotenv                    | 2.2.1        |
| excon                     | 0.57.1       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.12.2       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.11.0.1     |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| jwt                       | 1.5.6        |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.2.2        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.3        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.0.2        |
| google-api-client         | 0.12.0       |
| json                      | 1.8.6        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.3.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| slack-notifier            | 1.5.1        |
| terminal-notifier         | 1.8.0        |
| terminal-table            | 1.4.5        |
| tty-screen                | 0.5.0        |
| word_wrap                 | 1.0.0        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.1        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| xcpretty-travis-formatter | 0.0.4        |
| net-ssh                   | 4.1.0        |
| net-sftp                  | 2.1.2        |
| krausefx-shenzhen         | 0.14.11      |
| netrc                     | 0.11.0       |
| rest-client               | 2.0.2        |
</details>

*generated on:* **2017-07-24**
</details>

**Comment by jlubeck** [comment]
I should note that it is happening with and without bundle. On my local computer and on a remote CI server as well... so weird

**Comment by mpirri** [comment]
Interesting! Does the same thing happen if you directly invoke `bundle exec pod install --repo-update` on your project in Terminal?

Also, what is the line you have in your `Fastfile` to invoke Cocoapods?

Finally, it appears that your version of the `cocoapods` gem is rather old (version 0.2.0). While I don't necessarily think that is the problem, I just thought I'd mention it, in case it isn't what you intended.

Thanks for writing in, hopefully we'll get this figured out quickly!  🚀

**Comment by jlubeck** [comment]
Hey @mpirri !
I did notice the version 0.2.0 as well. That is definitely now what I want, but when I run bundle update, that is what I get!

This is the line in the Fastile
   ` cocoapods(repo_update: true)`
I tried without the repo_update and the same happens.

This is my Gemfile
```
gem 'fastlane'
gem 'cocoapods'
gem 'rest-client'
gem 'aws-sdk'
gem 'krausefx-shenzhen'
```

And yes, running `bundle exec pod install --repo-update` does the same thing.

Although running without the bundle exec works fine.

I understand that by trying this last bit it sounds more of problem from either cocoapods or bundle, but if you have any ideas of what I might try I would appreciate it!

**Comment by jlubeck** [comment]
Ok, I forced a version of Cocopods on the Gemfile

`gem 'cocoapods' , '~> 1.2.0'`

Now everything works fine. Any idea why this might have happened? Who should I let know, cocopods team or bundle team?

Feel free to close this one

Thanks for the direction on trying bundle on the cocopods alone! Don't know how I missed that

Thanks!

**Comment by mpirri** [comment]
Wow, so weird.  I played around with a similar combination of gems in a empty project directory.  Basically, if I had just this `Gemfile`, I'd get `cocoapods 0.2.0`:
```
source 'https://rubygems.org'
gem 'cocoapods'
gem 'rest-client'
gem 'fastlane'
```
Depending on if I remove `rest-client` or `fastlane` or both, I'd get different versions of `cocoapods`. I tried to understand why Bundler was making that choice of versions, but to no avail. I imagine there is a way to get `bundle` to explain why it is making the choices that it is making, but my Bundler expertise runs out just short of knowing how to do that.  😄 

Probably worth asking the Bundler folks about it (since the decision is being made as part of the dependency resolution), but I'm sure some `cocoapods` people might have ideas as well.

I'll close the issue now, thanks for the update!  🚀
